cms/models/placeholdermodel.py

```diff
@@ -299,4 +306,81 @@ def actions(self):
             self._actions_cache = getattr(field, 'actions', PlaceholderNoAction())
         return self._actions_cache
 
+    def get_cache_expiration(self, request, response_timestamp):
+        """
+        Returns the number of seconds (from «response_timestamp») that this
+        placeholder can be cached. This is derived from the plugins it contains.
+
+        This method must return: EXPIRE_NOW <= int <= MAX_EXPIRATION_IN_SECONDS
+
+        :type request: HTTPRequest
+        :type response_timestamp: datetime
+        :rtype: int
+        """
+        lang = getattr(request, 'lang', None)
+
+        min_ttl = MAX_EXPIRATION_TTL
+        for plugin_item in self.get_plugins(lang):
+            instance, plugin = plugin_item.get_plugin_instance()
```

Comment:
I was using DDT too, but rechecked things after seeing your results. I blew away my *.pyc files, and I get the results you see. (I did say it seemed logical! =)).

Anyway, digging into it a little, I think I've now got a better solution. a lot of great work is done in `cms.utils.plugins.downcast_plugins()` and I've now tweaked the code to leverage this to not re-query what we've already queried. The result is the performance I was seeing above, but with fresh `pyc` files =).